scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,336 @@
+        values by default.
+    irl_mode : bool, optional
+        If `True`, then compute SVD using IRL (implicitly restarted Lanczos)
+        mode.  Default is `False`.
+    kmax : int, optional
+        Maximal number of iterations / maximal dimension of Krylov subspace.
+        Default is ``5 * k``.
+    compute_u : bool, optional
+        If `True` (default) then compute left singular vectors, `u`.
+    compute_v : bool, optional
+        If `True` (default) then compute right singular vectors, `v`.
+    tol : float, optional
+        The desired relative accuracy for computed singular values.
+        If not specified, it will be set based on machine precision.
+    v0 : array_like, optional
+        Starting vector for iterations: must be of length ``A.shape[0]``.
+        If not specified, PROPACK will generate a starting vector.
+    full_output : bool, optional
+        If `True`, then return sigma_bound.  Default is `False`.
+    delta : float, optional
+        Level of orthogonality to maintain between Lanczos vectors.
+        Default is set based on machine precision.
+    eta : float, optional
+        Orthogonality cutoff.  During reorthogonalization, vectors with
+        component larger than `eta` along the Lanczos vector will be purged.
+        Default is set based on machine precision.
+    anorm : float, optional
+        Estimate of ``||A||``.  Default is `0`.
+    cgs : bool, optional
+        If `True`, reorthogonalization is done using classical Gram-Schmidt.
+        If `False` (default), it is done using modified Gram-Schmidt.
+    elr : bool, optional
+        If `True` (default), then extended local orthogonality is enforced
+        when obtaining singular vectors.
+    blocksize : int, optional
+        If computing `u` or `v` (left and right singular vectors,
+        respectively), `blocksize` controls how large a fraction of the
+        work is done via fast BLAS level 3 operations.  A larger blocksize
+        may lead to faster computation at the expense of greater memory
+        consumption.  `blocksize` must be ``>= 1``.  Default is `16`.
+    min_relgap : float, optional
+        The smallest relative gap allowed between any shift in IRL mode.
+        Default is `0.001`.  Accessed only if ``irl_mode=True``.
+    shifts : int, optional
+        Number of shifts per restart in IRL mode.  Default is determined
+        to satisfy ``k <= min(kmax-shifts, m, n)``.  Must be
+        >= 0, but choosing 0 might lead to performance degredation.
+        Accessed only if ``irl_mode=True``.
+    maxiter : int, optional
+        Maximum number of restarts in IRL mode.  Default is `1000`.
+        Accessed only if ``irl_mode=True``.
+
+    Returns
+    -------
+    u : ndarray
+        The top `k` left singular vectors, ``shape == (A.shape[0], 3)``,
+        returned only if ``compute_u=True``.
+    sigma : ndarray
+        The top `k` singular values, ``shape == (k,)``
+    vt : ndarray
```

Comment:
Pedantic, but `vh`?

Reply:
Doesn't bother me either way. I've never really had a use for complex linear algebra, so `vt` is fine with me, but yes, `vh` is more general.

Reply:
Changed this to make it consistent with the `return_singular_vector` parameter option.